### ISSUE TITLE
What happens when I change the conf_thres?

### ISSUE THREAD

**Issue by sakurasakura1996** [ultralytics/yolov5#7325]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, thank you very much for your work, I would like to ask: I used val.py to calculate the mAP of the training model on the test set and the result was 0.734, then I changed conf-thres from 0.001 to 0.1 and the mAP result changed to 0.742, can you tell me why this is, does the increase in mAP indirectly lead to a decrease in any metric. Increasing the confidence score, my understanding will increase the recall and decrease the accuracy, but the result is the opposite of what I understand
![image](https://user-images.githubusercontent.com/33887740/162106369-15eeaad4-5beb-4acd-ae5c-81a3149e56c7.png)
![image](https://user-images.githubusercontent.com/33887740/162106425-ca99dddb-e8d2-4fd7-af72-fec6cd6d9df7.png)

Looking forward to your reply, thank you!

**Comment by glenn-jocher** [comment]
@sakurasakura1996 --conf values above 0.001 are invalid as the code states. If you are not seeing this warning then update your code:
https://github.com/ultralytics/yolov5/blob/0ca85ed65f124871fa7686dcf0efbd8dc9699856/val.py#L369-L370

To update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

**Comment by sakurasakura1996** [comment]
> @sakurasakura1996 --conf values above 0.001 are invalid as the code states. If you are not seeing this warning then update your code:
> 
> https://github.com/ultralytics/yolov5/blob/0ca85ed65f124871fa7686dcf0efbd8dc9699856/val.py#L369-L370
> 
> To update:
> 
> * **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
> * **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
> * **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  [![Open In Colab](https://camo.githubusercontent.com/84f0493939e0c4de4e6dbe113251b4bfb5353e57134ffd9fcab6b8714514d4d1/68747470733a2f2f636f6c61622e72657365617263682e676f6f676c652e636f6d2f6173736574732f636f6c61622d62616467652e737667)](https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb) [![Open In Kaggle](https://camo.githubusercontent.com/a08ca511178e691ace596a95d334f73cf4ce06e83a5c4a5169b8bb68cac27bef/68747470733a2f2f6b6167676c652e636f6d2f7374617469632f696d616765732f6f70656e2d696e2d6b6167676c652e737667)](https://www.kaggle.com/models/ultralytics/yolov5)
> * **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image [![Docker Pulls](https://camo.githubusercontent.com/280faedaf431e4c0c24fdb30ec00a66d627404e5c4c498210d3f014dd58c2c7e/68747470733a2f2f696d672e736869656c64732e696f2f646f636b65722f70756c6c732f756c7472616c79746963732f796f6c6f76353f6c6f676f3d646f636b6572)](https://hub.docker.com/r/ultralytics/yolov5)

OK, thank you

**Comment by pderrenger** [comment]
@sakurasakura1996 you're welcome! If you have any more questions or need further assistance, feel free to ask.
